docs/design/ha_global_restarts.md

```diff
@@ -0,0 +1,96 @@
+# Global restarts for HA clusters
+
+## Metadata
+
+* Authors: @johscheuer
+* Created: 2022-10-03
+* Updated: 2022-10-03
+
+## Background
+
+The current implementation of normal restarts (not version upgrades) are not synchronized between different operators for HA FDB clusters.
+This has multiple drawbacks if a user runs a HA FDB cluster, independent if those are managed by the same operator or not.
+One example of the underlying issue is [that a restart in one cluster will delay the restart in another cluster](https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1361).
+
+The current limitations of this decoupled process are:
+
+- Rolling out a new knob to an HA cluster takes at least `minimumUptime * (number of clusters - 1)` seconds.
+- Coordinator changes that are initiated by other operator, e.g. because an fdbserver will be unavailable for a short moment during th restart, can delay the rollout further (the new cluster file must be synced).
+- We will see at least `number of clusters` recoveries to rollout a knob instead of `1`.
+
+## General Design Goals
+
+In this design we want to propose a new mechanism to rollout knob changes for HA clusters to reduce the number of restarts to one and have the restart coordinated across different operators.
+
+## Current Implementation
+
+The current implementation only waits until all process groups in the current cluster configuration are "ready" to restart.
+Before the operator tries to restart processes it waits until the new configuration is synced to all affected processes that have the `IncorrectCommandLine` condition.
+Once all processes have the new configuration the operator triggers a `fdbcli --exec 'kill; kill ...'` command to restart those fdbserver process managed by this operator instance.
```

Comment:
is there mechanism in k8s ecosystem that can bounce pods or processes in pods (roughly) simultaneously?